scripts/ci/pre_commit/pre_commit_check_pre_commit_hook_names.py

```diff
@@ -50,16 +57,112 @@ def main() -> int:
     return retval
 
 
-def get_errors(content, max_length):
+def get_errors_and_ids(content, max_length):
     errors = []
+    ids = set()
     for repo in content['repos']:
         for hook in repo['hooks']:
+            if 'id' in hook:
+                id = hook['id']
+                ids.add(id)
             if 'name' not in hook:
                 continue
             name = hook['name']
             if len(name) > max_length:
                 errors.append(name)
-    return errors
+    return errors, list(ids)
+
+
+def render_template(
+    searchpath: Path,
+    template_name: str,
+    context: Dict[str, Any],
+    extension: str,
+    autoescape: bool = True,
+    keep_trailing_newline: bool = False,
+) -> str:
+    """
+    Renders template based on its name. Reads the template from <name>_TEMPLATE.md.jinja2 in current dir.
+    :param template_name: name of the template to use
+    :param context: Jinja2 context
+    :param extension: Target file extension
+    :param autoescape: Whether to autoescape HTML
+    :param keep_trailing_newline: Whether to keep the newline in rendered output
+    :return: rendered template
+    """
+    import jinja2
+
+    template_loader = jinja2.FileSystemLoader(searchpath=searchpath)
+    template_env = jinja2.Environment(
+        loader=template_loader,
+        undefined=jinja2.StrictUndefined,
+        autoescape=autoescape,
+        keep_trailing_newline=keep_trailing_newline,
+    )
+    template = template_env.get_template(f"{template_name}_TEMPLATE{extension}.jinja2")
+    content: str = template.render(context)
+    return content
+
+
+def get_precommit_jinja_context(pre_commit_ids: List) -> Dict[str, List]:
+    context: Dict[str, List] = {"PRE_COMMIT_IDS": pre_commit_ids}
+    return context
+
+
+def get_target_precommit_parent_dir():
+    airflow_source = Path.cwd()
+    pre_commit_ids_file = Path(airflow_source, "dev", "breeze", "src", "airflow_breeze")
+    return pre_commit_ids_file
+
+
+def get_breeze_pyproject_toml_dir():
+    airflow_source = Path.cwd()
+    breeze_pyproject_toml_file = Path(airflow_source, "dev", "breeze")
+    return breeze_pyproject_toml_file
+
+
+@lru_cache(maxsize=None)
+def black_mode():
```

Comment:
NICE!

Reply:
I think you could add nice unit test for that one (to check if you have the right values retrieved"

```
assert mode.target_versions == ....
```

Reply:
Yeah... I changed my mind.  I think actually this one does not need unit tests - those pre-commit.py scripts are executed with every commit, so there is no point in addint extra tests for those - the pre-commits are actually "testing themselves" in a way.

Reply:
hmm ok. could you tell me how to decide if the code that has to be covered in the test or not? Like the above one, you told me it wouldn't require tests as it executes with every commit. Does those kinds of pre-commits tests are not necessary to be covered by tests? @potiuk

Reply:
I think it's really the matter of "is it actually executed with every PR". 

Pre-commits are executed always with every PR (nearly). So if something breaks, we will likely find out quickly. Unit tests are really there to execute the code often and assert if it still behaves as it should. 

Pre-comit is slighlty different because we do not 'formally" assert if they do what they should do - this is small risk, becuase the precommit might simply "do it wrongly". But that's an acceptable risk. Pre-commits are there to prevent us from making mistakes that we shoudl also caught during the code review (basically - pre-commits automate big part of the reviewer's job really), so we are likely to catch any of such mistakes during code review.